### PR TITLE
Proposal for ActionCreator Interface

### PR DESCRIPTION
**Pull request by james-thimont-bcgdv**
Constrained protocol associatedtypes to StateType

### CODE REVIEW

File: ReSwift/CoreTypes/ActionCreator.swift [hunk 1]
```diff
@@ -10,9 +10,18 @@ import Foundation
 
 public protocol ActionCreator {
     
-    associatedtype State
+    associatedtype State: StateType
+    /**
+     Generates one or more new actions from the current state.  May be implemented asynchronously or synchronously
+     - parameter actionBlock This may be called more than once with different actions
+     */
+    func createActions(currentState state: State, actionBlock:(Action -> Void))
```
Comment (jonathanellis) [inline ReSwift/CoreTypes/ActionCreator.swift]: Does the ActionCreator not need a reference to a dispatcher? This is OK if the store is a global var but we may want to move away from that

Comment (james-thimont-bcgdv) [inline ReSwift/CoreTypes/ActionCreator.swift]: @jonathanellis ActionCreator doesn't dispatch the actions, it just creates them.  
Some other object is responsible for dispatching them

Comment (jonathanellis) [inline ReSwift/CoreTypes/ActionCreator.swift]: Also why is Action a parameter of the actionBlock? Sure it should be () -> Action? as the ActionCreator can choose to return an Action to be performed immediately, or can defer and create actions at a later date asynchronously.

Can we not then redefine the actionBlock as (State -> Action?) and get rid of currentState?

**Comment by jonathanellis** [comment]
👍

**Comment by jonathanellis** [comment]
@james-thimont-bcgdv can you merge? thx

**Comment by jonathanellis** [comment]
can you also version bump your ReSwift pod so we don't end up with dependency versioning issues

**Comment by jonathanellis** [comment]
👍

**Pull request merged**